File: OpenEdXMobile/src/test/java/org/edx/mobile/test/BaseTestCase.java
package org.edx.mobile.test;

import android.content.Context;

import androidx.test.core.app.ApplicationProvider;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

import org.edx.mobile.Injector;
import org.edx.mobile.logger.Logger;
import org.edx.mobile.test.util.TimeUtilsForTests;
import org.edx.mobile.util.Config;
import org.junit.After;
import org.junit.Before;
import org.junit.Ignore;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.TimeZone;

@Ignore
public abstract class BaseTestCase extends BaseTest {
    protected final Logger logger = new Logger(getClass().getName());

    protected Context context;
    protected Config config;

    @Before
    public void setUp() throws Exception {
        context = ApplicationProvider.getApplicationContext();
        config = new Config(generateConfigProperties());

        Injector injector = new Injector(config);
        inject(injector);
        // Set time zone to a constant value to make time-based tests predictable
        TimeZone.setDefault(TimeUtilsForTests.DEFAULT_TIME_ZONE);
        print("Started Test Case: " + getClass().getName());
    }

    /**
     * subclass should inject the properties marked as @Inject
     */
    protected void inject(Injector injector) throws Exception {
    }

    protected JsonObject generateConfigProperties() throws IOException {
        // Generate default config properties for subclasses to customize
        try (InputStream in = context.getAssets().open("config/config.json")) {
            return new JsonParser().parse(new InputStreamReader(in)).getAsJsonObject();
        }
    }


    @After
    public void tearDown() throws Exception {
        print("Finished Test Case: " + getClass().getName());
    }

    protected void print(String msg) {
        System.out.println(msg);
        logger.debug(msg);
    }
}

File: OpenEdXMobile/src/sharedTest/java/org/edx/mobile/base/BaseTestCase.kt
package org.edx.mobile.base

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import com.google.gson.JsonObject
import com.google.gson.JsonParser
import org.edx.mobile.logger.Logger
import org.edx.mobile.test.util.TimeUtilsForTests
import org.edx.mobile.util.Config
import org.junit.After
import org.junit.Before
import org.junit.Ignore
import java.io.IOException
import java.io.InputStreamReader
import java.util.*

@Ignore
abstract class BaseTestCase : BaseTest() {

    @JvmField
    protected val logger = Logger(javaClass.name)

    protected lateinit var context: Context

    protected lateinit var config: Config

    @Before
    @Throws(Exception::class)
    open fun setUp() {
        context = ApplicationProvider.getApplicationContext()
        config = Config(generateConfigProperties())
        val injector = Injector(config)
        inject(injector)
        // Set time zone to a constant value to make time-based tests predictable
        TimeZone.setDefault(TimeUtilsForTests.DEFAULT_TIME_ZONE)
        print("Started Test Case: " + javaClass.name)
    }

    /**
     * subclass should inject the properties marked as @Inject
     */
    @Throws(Exception::class)
    protected open fun inject(injector: Injector?) {
    }

    @Throws(IOException::class)
    protected open fun generateConfigProperties(): JsonObject? {
        // Generate default config properties for subclasses to customize
        context.assets?.open("config/config.json").use { inputStream ->
            return JsonParser().parse(
                InputStreamReader(inputStream)
            ).asJsonObject
        }
    }

    @After
    @Throws(Exception::class)
    open fun tearDown() {
        print("Finished Test Case: " + javaClass.name)
    }

    protected fun print(msg: String?) {
        println(msg)
        logger.debug(msg)
    }
}
